File: matlab/project1/BaxterClass.m
% METADATA OF THE FILE
%{ 
    File: BaxterClass.m
    Class: Industrial Robotics 
    Author: Elkin Javier Guerra Galeano
    Created on 09-16-2020, 16:14
%}

% Definimos la clase BAXTER 
classdef BaxterClass
    %% Definimos los atributos de baxter
    properties
        % Medidas de los brazos de baxter
        l0 = 0.27035;
        l1 = 0.06900;
        l2 = 0.36435;
        l3 = 0.06900;
        l4 = 0.37429;
        l5 = 0.01000;
        %l5 = 0;
        l6 = 0.36830;
        % Medidas adicionales 
        L = 0.27800;
        h = 0.06400;
        H = 1.10400;
    end
    
    %% Definimos los metodos de baxter
    methods
        % Metodo para calcular la FPK 
        function [Two_t] = FPK(obj,dof,arm,ndof)
            Two_t = FPK(dof,[obj.l0;obj.l1;...
            obj.l2;obj.l3;obj.l4;obj.l5;obj.l6;obj.L;obj.h;obj.H],arm,ndof);
        end 
        % Metodo para calcular la IPK (6 dof)
        function [vdof] = IPK(obj,FPK,arm,type)
            vdof = IPK(FPK,[obj.l0;obj.l1;obj.l2;...
            obj.l3;obj.l4;obj.l6;obj.L;obj.h;obj.H],arm,type);
        end
        % Metodo para calcular la IPK (7 dof)
        function [vdof] = IPK_7dof(obj,FPK,arm,dof_0,iter,delta,tol)
            vdof = IPK_7dof(FPK,[obj.l0;obj.l1;obj.l2;...
            obj.l3;obj.l4;obj.l6;obj.L;obj.h;obj.H],arm,dof_0,iter,delta,tol);
        end
        % Metodo para calcular los puntos del cuerpo de baxter
        function [P_list] = Puntos_Baxter(obj,dof,arm)
            P_list = Puntos_Simulacion(dof,[obj.l0;obj.l1;...
            obj.l2;obj.l3;obj.l4;obj.l5;obj.l6;obj.L;obj.h;obj.H],arm);
        end
    end
end

File: python/project_1_move_baxter/BaxterClass.py
#!/usr/bin/env python
# Importamos modulos importantes
import IPK
import FPK
import numpy as np
import math

# Definimos la clase BAXTER


class baxterClass():
    # Definimos los atributos de baxter
    def __init__(self):
        # Medidas de los brazos de baxter
        self.l0 = 0.27035
        self.l1 = 0.06900
        self.l2 = 0.36435
        self.l3 = 0.06900
        self.l4 = 0.37429
        self.l5 = 0.01000
        self.l6 = 0.36830
        # Medidas adicionales
        self.L = 0.27800
        self.h = 0.06400
        self.H = 1.10400# - 0.0601

    # Definimos los metodos de baxter
    def fpk(self, dof, arm, ndof):
        Two_t = FPK.FPK(dof, [self.l0, self.l1,
                              self.l2, self.l3, self.l4, self.l5, self.l6, self.L, self.h, self.H], arm, ndof)
        return Two_t

    def ipk(self, FPK, arm, type):
        vdof = IPK.IPK(FPK, [self.l0, self.l1, self.l2,
                             self.l3, self.l4, self.l6, self.L, self.h, self.H], arm, type)
        return vdof


if __name__ == '__main__':
    pass
